wallet/address.js

```diff
@@ -87,23 +87,22 @@ async function deriveAddressAndSecret(rootSecretString, childIndex) {
     addressPayload = new Buffer(0)
     addressAttributes = new Map()
     derivedSecretString = rootSecretString
-    addressRoot = new Buffer(getAddressRoot(derivedSecretString, addressPayload), 'hex')
   } else {
     // the remaining addresses
     const hdPassphrase = await deriveHDPassphrase(rootSecretString)
     derivedSecretString = deriveSK(rootSecretString, childIndex)
```

Comment:
please move this line to the end of the block so that it is consistent with the previous if branch